atomic_reactor/plugins/post_compare_components.py

```diff
@@ -0,0 +1,105 @@
+"""
+Copyright (c) 2017 Red Hat, Inc
+All rights reserved.
+
+This software may be modified and distributed under the terms
+of the BSD license. See the LICENSE file for details.
+"""
+from atomic_reactor.plugin import PostBuildPlugin
+from atomic_reactor.constants import (PLUGIN_COMPARE_COMPONENTS_KEY,
+                                      PLUGIN_FETCH_WORKER_METADATA_KEY)
+
+
+class CompareComponentsPlugin(PostBuildPlugin):
+    """
+    Compare components from each worker build and verify the same version was
+    on each worker.
+    """
+
+    key = PLUGIN_COMPARE_COMPONENTS_KEY
+    is_allowed_to_fail = False
+
+    def rpm_compare(self, a, b):
+        """
+        Compare rpm component version 'a' with 'b'.
+        'name' is implied equal as it is the key() lookup
+        """
+        if a['version'] == b['version'] and \
+           a['release'] == b['release'] and \
+           a['type'] == b['type'] and \
```

Comment:
We're already comparing 'type' in the main loop.

Reply:
It seems to belong here, and not in the main loop.

Reply:
My counter argument is 
pkgA['type'] = 'rpm' and pkgB['type'] = 'foo'.  pkgA makes it to master_comp list (new pkg), but pkgB triggers 'unknown' type because it fails to match type == 'rpm'.

So the warning is misleading one to think pkgB is new and incorrectly labeled when one can conclude it is a 'version' mismatch. 

Now swap the types pkgA = foo and pkgB = rpm, you get the version mismatch error as expected because pkgA is blindly added to master_comp list.

This is why I put the 'type' check in the 'main' function.  I thought it would explicitly catch the bad scenarios as opposed to relying on a random order to determine what warning is generated.

Feel free to disagree and I can make the changes.

Reply:
I'm not so sure we should even be comparing components of different types, just like we don't compare components with different names.

I tihnk 'type' and 'name' should be used together as the key for comparison.